horizons/ai/aiplayer/settlementmanager.py

```diff
@@ -228,13 +228,15 @@ def _set_taxes_and_permissions_prefix(self, prefix):
 		pioneer_taxes = getattr(self.personality, '{}_pioneer_taxes'.format(prefix))
 		settler_taxes = getattr(self.personality, '{}_settler_taxes'.format(prefix))
 		citizen_taxes = getattr(self.personality, '{}_citizen_taxes'.format(prefix))
-		merchants_taxes = getattr(self.personality, '{}_merchants_taxes'.format(prefix))
+		#merchants_taxes = getattr(self.personality, '{}_merchants_taxes'.format(prefix))
 		sailor_upgrades = getattr(self.personality, '{}_sailor_upgrades'.format(prefix))
 		pioneer_upgrades = getattr(self.personality, '{}_pioneer_upgrades'.format(prefix))
 		settler_upgrades = getattr(self.personality, '{}_settler_upgrades'.format(prefix))
-		citizen_upgrades = getattr(self.personality, '{}_citizen_upgrades'.format(prefix))
-		self._set_taxes_and_permissions(sailor_taxes, pioneer_taxes, settler_taxes, citizen_taxes, merchants_taxes,
-			sailor_upgrades, pioneer_upgrades, settler_upgrades, citizen_upgrades)
+		#citizen_upgrades = getattr(self.personality, '{}_citizen_upgrades'.format(prefix))
+		merchants_taxes = None
+		citizen_upgrades = None
+		#self._set_taxes_and_permissions(sailor_taxes, pioneer_taxes, settler_taxes, citizen_taxes, merchants_taxes,
```

Comment:
I don't think commenting out the entire call is a good idea. Notice that also ignores *_upgrades and *_taxes for valid tiers (sailors, pioneers, etc.).

`getattr()` also takes a default value as the last argument so that could be used to indicate that `merchants_taxes` and `citizen_upgrades` is missing / not implemented.

Reply:
You're definitely right.  I'm confused on how that occurred, as that is the entire point of setting the two values to None was to make it still possible to do that call. (Will test it again, in the rare cause calling it with None was breaking it.)

Reply:
It was intentonal that I commented it out. There was an attempt with them set to None, (which is why they're set that way) but that crashes (as follows). Note: this isn't a complete fix, just one that gets it not-crashing.
`Traceback (most recent call last):
  File "./run_uh.py", line 385, in <module>
    main()
  File "./run_uh.py", line 127, in main
    ret = horizons.main.start(options)
  File "/home/fu-fu/source/unknown-horizons/horizons/main.py", line 255, in start
    horizons.globals.fife.run()
  File "/home/fu-fu/source/unknown-horizons/horizons/engine/engine.py", line 280, in run
    self.loop()
  File "/home/fu-fu/source/unknown-horizons/horizons/engine/engine.py", line 297, in loop
    f()
  File "/home/fu-fu/source/unknown-horizons/horizons/timer.py", line 112, in check_tick
    f(self.tick_next_id)
  File "/home/fu-fu/source/unknown-horizons/horizons/scheduler.py", line 95, in tick
    callback.callback()
  File "/home/fu-fu/source/unknown-horizons/horizons/util/python/callback.py", line 47, in __call__
    return self.callback(*self.args, **self.kwargs)
  File "/home/fu-fu/source/unknown-horizons/horizons/ai/aiplayer/__init__.py", line 317, in tick
    self.handle_settlements()
  File "/home/fu-fu/source/unknown-horizons/horizons/ai/aiplayer/__init__.py", line 364, in handle_settlements
    settlement_manager.refresh_taxes_and_upgrade_permissions()
  File "/home/fu-fu/source/unknown-horizons/horizons/ai/aiplayer/settlementmanager.py", line 376, in refresh_taxes_and_upgrade_permissions
    self._set_taxes_and_permissions_prefix('early')
  File "/home/fu-fu/source/unknown-horizons/horizons/ai/aiplayer/settlementmanager.py", line 239, in _set_taxes_and_permissions_prefix
    sailor_upgrades, pioneer_upgrades, settler_upgrades, citizen_upgrades)
TypeError: _set_taxes_and_permissions() takes 8 positional arguments but 10 were given
AL lib: (EE) alc_cleanup: 1 device not closed`

Reply:
A better fix (I assume) is now provided, altering the method call to match the function.